File: src/native/printer-plugin/drivers/generic-escpos.driver.test.ts
import { describe, it, expect } from 'vitest'
import { GenericEscPosDriver } from './generic-escpos.driver'

describe('GenericEscPosDriver', () => {
  it('starts the byte stream with the ESC @ initialize command', () => {
    const bytes = new GenericEscPosDriver().buildTestPrintBytes()
    expect(Array.from(bytes.slice(0, 2))).toEqual([0x1b, 0x40])
  })

  it('ends the byte stream with the GS V 0 full-cut command', () => {
    const bytes = new GenericEscPosDriver().buildTestPrintBytes()
    expect(Array.from(bytes.slice(-3))).toEqual([0x1d, 0x56, 0x00])
  })

  it('contains the sample bill text from the project spec', () => {
    const bytes = new GenericEscPosDriver().buildTestPrintBytes()
    const text = new TextDecoder().decode(bytes)
    expect(text).toContain('NDT Bubble Tea')
    expect(text).toContain('Order #1001')
    expect(text).toContain('TOTAL')
    expect(text).toContain('$8.50')
    expect(text).toContain('Thank You')
  })
})

File: android/app/src/test/java/com/ndtcore/pos/printer/driver/GenericEscPosDriverTest.kt
package com.ndtcore.pos.printer.driver

import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class GenericEscPosDriverTest {
    @Test
    fun `buildTestPrintBytes starts with ESC init command`() {
        val bytes = GenericEscPosDriver().buildTestPrintBytes()
        assertArrayEquals(byteArrayOf(0x1B, 0x40), bytes.copyOfRange(0, 2))
    }

    @Test
    fun `buildTestPrintBytes ends with full cut command`() {
        val bytes = GenericEscPosDriver().buildTestPrintBytes()
        assertArrayEquals(byteArrayOf(0x1D, 0x56, 0x00), bytes.copyOfRange(bytes.size - 3, bytes.size))
    }

    @Test
    fun `buildTestPrintBytes contains the sample bill text`() {
        val text = String(GenericEscPosDriver().buildTestPrintBytes(), Charsets.US_ASCII)
        assertTrue(text.contains("NDT Bubble Tea"))
        assertTrue(text.contains("TOTAL"))
        assertTrue(text.contains("Thank You"))
    }
}
